crates/c-api/CMakeLists.txt

```diff
@@ -100,6 +99,9 @@ find_program(WASMTIME_CARGO_BINARY cargo)
 if(NOT WASMTIME_CARGO_BINARY)
     message(FATAL_ERROR [["cargo" was not found. Ensure "cargo" is in PATH. Aborting...]])
 endif()
+# Note: this line will *always* create ${CMAKE_CURRENT_SOURCE_DIR}/artifact
+# during the configure stage of cmake, and there does not seem to be a way to
+# disable this behavior. This prevents cargo package from verifying the package.
```

Comment:
Should the `BINARY_DIR` configuration be dropped entirely? That should probably be somewhere within `CMAKE_CURRENT_BINARY_DIR` if it's still being used to avoid modifying the source tree.